iree/compiler/Dialect/Flow/Transforms/DispatchLinalgOnTensors.cpp

```diff
@@ -503,6 +504,12 @@ struct TileAndDistributeOnTensorsPattern
     SmallVector<Value, 4> count = llvm::to_vector<4>(
         llvm::map_range(linalgOp.createLoopRanges(rewriter, loc),
                         [](Range r) { return r.size; }));
+    // XXX: Special treatment for convolution, which have more than 3 parallel
```

Comment:
XXX?

Reply:
Actually I don't quite know what to use here. It's not a TODO given this is what we'd like it to be. I personally feel it should be stronger than a NOTE. I saw XXX used in some codebase so used it as a placeholder for now. Any suggestions?

Reply:
Or maybe I'm just use NOTE here.

Reply:
This would not be the place to do this. You need to change the tile sizes used. The tileSize of 0 means that loop is not-tiled. SO you set that below when you are adding the patterns. Here you probably need to change the count computation to drop the loop ranges for the untiled loops and only take the outer parallel loops that are tiled.

Reply:
I think we need both. Here it's for the real workload passed to `flow.dispatch.workgroups` op itself and it's for host side compilation. Setting tile size to zero for the first dimension is also needed and that's for device side compilation.